src/utils/proxyMap.ts

```diff
@@ -0,0 +1,113 @@
+import { proxy } from 'valtio'
+
+type KeyValRecord<K, V> = {
+  key: K
+  value: V
+}
+
+type InternalProxyMap<K, V> = Map<K, V> & {
+  data: KeyValRecord<K, V>[]
+  toJSON: object
+}
+
+/**
+ * proxyMap
+ *
+ * This is to create a proxy which mimic the native Map behavior.
+ * The API is the same as Map API
+ *
+ * @example
+ * import { proxyMap } from 'valtio/utils'
+ * const state = proxyMap([["key", "value"]])
+ *
+ * //can be used inside a proxy as well
+ * const state = proxy({
+ *   count: 1,
+ *   map: proxyMap()
+ * })
+ */
+export const proxyMap = <K, V>(
+  entries: Iterable<readonly [K, V]> | null = []
+): Map<K, V> => {
+  const map = proxy<InternalProxyMap<K, V>>({
+    data:
+      entries === null
```

Comment:
never becomes true.

Reply:
It does when we do `proxyMap(null)`. The default parameter `[]` is used only if the provided value is `undefined` or  if no value is passed to `proxyMap`

Reply:
My bad again!


---
This is just a nit, but since I got confused, here's my preference (not a strong suggestion):

```js
export const proxyMap = <K, V>(
  entries?: Iterable<readonly [K, V]> | null
): Map<K, V> => {
  const map = proxy<InternalProxyMap<K, V>>({
    data: Array.from(entries || []).map((v) => ({ key: v[0], value: v[1] })),
```

And for `proxySet`:
```js
export const proxySet = <T>(initialValues?: Iterable<T> | null): Set<T> => {
  const set: InternalProxySet<T> = proxy({
    data: Array.from(new Set(initialValues)),
```

Reply:
actually, I prefer your suggestion. It's way cleaner 👌. For `proxySet` do you want me to open a separate PR?

Reply:
Cool. I often forget about the 2nd arg too.

If the change to `proxySet` is just that one, I think we can do it in this PR.